packages/grid/x-data-grid/src/models/api/gridColumnApi.ts

```diff
@@ -12,6 +12,12 @@ export interface GridColumnApi {
    * @returns {{GridStateColDef}} The [[GridStateColDef]].
    */
   getColumn: (field: string) => GridStateColDef;
+  /**
+   * Returns the [[GridColDef]]s for the given `fields`.
+   * @param {string[]} fields Array of column fields.
+   * @returns {{GridStateColDef[]}} Array of [[GridStateColDef]]s.
+   */
+  getColumns: (fields: string[]) => GridStateColDef[];
```

Comment:
Where is this used?

Reply:
I did use this here:
https://github.com/mui/mui-x/blob/9ef911a6e036a69f8a03c398a76fe8ffe7e7afd8/packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnSpanning.ts#L96-L112

I did plan to also change the API here to receive `fields` in the options as well.
The problem is that it is used internally as well.

So I am currently looking to create a wrapper around `calculateColSpan` that is public and accepts `fields` as an option instead of `columns`

The idea looks like this (not taking React hooks into account):
```ts
const calculateColSpan = ({fields, ...rest}) => {
  const columns = apiRef.current.getColumns(fields);
  return internalCalculateColSpan({...rest, columns})
```

WDYT @romgrk ?

Reply:
Nevermind ... just noticed it's behind the `mui.api_private` symbol!